### ISSUE TITLE
Histogram of equal values returns Infinity

### ISSUE THREAD

**Issue by zinkkrysty** [jtprince/histogram#9]
```
irb(main):022:0> [12400, 12400].histogram(:sturges)
=> [[Infinity, Infinity], [2.0, 0.0]]
```

**Comment by jtprince** [comment]
I'll take a look and try to patch that up, probably sometime this weekend.  I've started to deal properly with some edge cases, but clearly they are not all sorted out.  Thanks for the report.

**Comment by jtprince** [comment]
Explicitly set bin size to 1 for all arrays with the same value if explicit bin size or values not given .  That deals with this case (and many other similar cases to it), and seems like the right thing to do when determining proper bin size.  That would set the bin size in your example to 1.

In addition, I added code to handle the case where there are many bins but the values are all the same (say, if you explicitly set the bin size to 2:     [1, 1].histogram(2)  ).  In this case, the bins are set to be the same value, which is the best solution amongst all solutions I can imagine.

released as histogram 0.2.4.1 to rubygems.org
